### PR TITLE
Update contributing guide

### PR DESCRIPTION
**Pull request by michael-valdron**
### What does this PR do?:
_Summarize the changes. Are any stacks or samples added or updated?_

Updates contributing guide and supporting documents with up to date information and layout.

Additionally, adds missing `DCO` file into repository.

### Which issue(s) this PR fixes:
_Link to github issue(s)_

fixes devfile/api#1347

### PR acceptance criteria:

- [X] Contributing guide
_Have you read the [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) and followed its instructions?_
- [x] Test automation
_Does this repository's tests pass with your changes?_
- [X] Documentation
_Does any documentation need to be updated with your changes?_
- [ ] Check Tools Provider
_Have you tested the changes with existing tools, i.e. Odo, Che, Console? (See [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) on how to test changes)_

### How to test changes / Special notes to the reviewer:

### CODE REVIEW

**Comment by michael-valdron** [comment]
> Check Tools Provider

N/A

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,19 +1,59 @@
...
+
+Once you set your user.name and user.email in your git config, you can sign your commit automatically with `git commit -s`.
+
+## How to Contribute:
+
+### Issues
+
+For issues relating to a specific devfile stack in this repository, please reach out to the devfile stack maintainer to determine where to open an issue.
+
+For issues relating to the hosted devfile registry service (<https://registry.devfile.io>), or devfile registries in general, please use the [devfile/api](https://github.com/devfile/api/) issue tracker for opening issues. Apply the `area/registry` label to registry issues for better visibility.
```
Comment (thepetk) [inline CONTRIBUTING.md:35]:
```suggestion
For issues relating to the hosted devfile registry service (<https://registry.devfile.io>), or devfile registries in general, please use the [devfile/api](https://github.com/devfile/api/issues) issue tracker for opening issues. Apply the `area/registry` label to registry issues for better visibility.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,19 +1,59 @@
...
+
+### Submitting Pull Request
+
+When you think the code is ready for review, create a pull request and link the issue associated with it.
+
+Owners of the repository will watch out for new PRs and provide reviews to them.
```
Comment (thepetk) [inline CONTRIBUTING.md:41]: Might worth mentioning the registry stack process review section?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,19 +1,59 @@
...
 The following are required to build the devfile index container image containing your stack or sample:
 
-- Docker 17.06 or later
+- Docker or Podman
 - Git
 
-## Instructions
+### Instructions
 
 1. Open an issue in the [devfile/api](https://github.com/devfile/api) repo to track adding a new stack or sample
```
Comment (thepetk) [inline CONTRIBUTING.md:58]: I think this is also mentioned above in the `How to Contribute` > `Issues` section. Apart from that, as a reader is not clear to me what the instructions is for? (e.g is for "how I could contribute" or something else)

**Review by thepetk — COMMENTED** [review]
Left some minor comments :)

File: CONTRIBUTING.md (hunk 3 above)
Comment (michael-valdron) [reply CONTRIBUTING.md]: @thepetk 

> I think this is also mentioned above in the `How to Contribute` > `Issues` section.

Referred back the the `Issues` section for this step.

> Apart from that, as a reader is not clear to me what the instructions is for? (e.g is for "how I could contribute" or something else)

Yeah I agree this wasn't very clear, how does this revision sound? https://github.com/devfile/registry/pull/421/commits/08bef5147547ace067d72fecd03d3a32ad41f631

**Review by thepetk — APPROVED** [review]
/lgtm

**Pull request merged**